lib/hyperswitch.js

```diff
@@ -287,32 +287,32 @@ HyperSwitch.prototype._request = function(req, options) {
     // (at least at the top level) under our feet.
     var childReq = cloneRequest(req);
 
+    // Look up the route in the tree.
     var match = this._priv.router.route(childReq.uri);
-    var methods = match && match.value && match.value.methods;
-    var handler = methods && (
-            (self._rootReq && self._rootReq.method === 'head' && methods.head)
-                || methods[childReq.method]
-                || methods.all);
-    if (!handler &&
-            (req.method === 'head'
-            || self._rootReq && self._rootReq.method === 'head')) {
-        handler = methods && methods.get;
-    }
-
-    if (match && !handler
-            && childReq.method === 'get'
-            && childReq.uri.path[childReq.uri.path.length - 1] === '') {
-        // A GET for an URL that ends with /: return a default listing
-        if (!match.value) { match.value = {}; }
-        if (!match.value.path) { match.value.path = '_defaultListingHandler'; }
-        handler = function(hyper, req) {
-            return self.defaultListingHandler(match, hyper, req);
-        };
-    }
-
+    var handler;
     if (match) {
         childReq.params = match.params;
         self._checkInternalApiRequest(childReq);
+
+        // Find a handler.
+        var methods = match && match.value && match.value.methods || {};
+        handler = methods[childReq.method] || methods.all;
+        if (!handler
+                && (childReq.method === 'head'
+                    || self._rootReq && self._rootReq.method === 'head')) {
+            handler = methods && methods.get;
+        }
+
+        if (!handler
+                && childReq.method === 'get'
+                && childReq.uri.path[childReq.uri.path.length - 1] === '') {
+            // A GET for an URL that ends with /: return a default listing
+            if (!match.value) { match.value = {}; }
+            if (!match.value.path) { match.value.path = '_defaultListingHandler'; }
+            handler = function(hyper, req) {
+                return self.defaultListingHandler(match, hyper, req);
+            };
+        }
     }
 
     if (handler) {
```

Comment:
I'm all for focusing on the normal case, but the more lines we have for that case, the less legible the overall code structure is.